src/whylogs/core/metrics/model_metrics.py

```diff
@@ -96,20 +96,21 @@ def merge(self, other):
         """
         if other is None:
             return self
-        if other.confusion_matrix is None and other.regression_metrics is None:
-            # TODO: return a copy instead
-            return self
-        if self.confusion_matrix is None and self.regression_metrics is None:
-            return other
-
-        if self.model_type is None or other.model_type is None:
-            model_type = ModelType.UNKNOWN
-        elif other.model_type != self.model_type:
-            model_type = ModelType.UNKNOWN
+       
+        model_type =ModelType.UNKNOWN
+        if (self.model_type not in (ModelType.REGRESSION, ModelType.CLASSIFICATION)):
```

Comment:
because they could pass a none, or something else. we need to make a change for a schema validation

Reply:
trying to avoid exceptions for now, but will add one once we clean the validation

Reply:
we need to take into account the fact we may start with an empty dataset profile with default values, like Unknow during a merge train.

Reply:
In that case we should break the branching logic explicitly (merging against an empty object) vs normal merging code path

Reply:
I think this would be easier to read and have the same result,
```
        model_type = self.model_type
        if model_type not in (ModelType.REGRESSION, ModelType.CLASSIFICATION):
            model_type = other.model_type 
            if model_type not in (ModelType.REGRESSION, ModelType.CLASSIFICATION):
                model_type = = ModelType.UNKNOWN
```

Reply:
done  in 
48a7c61